### ISSUE TITLE
Crash on editor exit

### ISSUE THREAD

**Issue by kosz78** [godotengine/godot#12946]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10, Godot 3 build from c1855dcff14a118828de16e850b0698dc76b25e9

**Issue description:**
Periodically crashes on editor exit with the following state:
![image](https://user-images.githubusercontent.com/3992195/32834112-e910e476-ca33-11e7-8180-a622240410fb.png)

**Comment by spusnei-pbsc** [comment]
related to #12813

**Comment by reduz** [comment]
If anyone wants to debug this on windows, feel welcome

**Comment by chongdashu** [comment]
Not able to reproduce so far with 64-bit windows and 64-bit Godot; going to try with 32-bit Godot

**Comment by chongdashu** [comment]
OK, managed to reproduce this when using 64-bit windows and 32-bit Godot.

**Comment by chongdashu** [comment]
Seems to be crashing because of 
```
ImageTexture::~ImageTexture() {

	VisualServer::get_singleton()->free(texture);
}
```

Where `VisualServer::get_singleton()` is returning null.
![image](https://user-images.githubusercontent.com/1178121/33858859-4efad3ba-dec9-11e7-9f46-666d18579a6c.png)

**Comment by chongdashu** [comment]
This should be fixed in 0f223783dca, if anyone would like to confirm

**Comment by akien-mga** [comment]
Fixed by #14860.
